libebpfdiscovery/src/DiscoveryBpf.cpp

```diff
@@ -134,13 +134,31 @@ void DiscoveryBpf::attachSyscallProbes() {
 	skel->links.kprobeSysClose = attachKprobe(skel->progs.kprobeSysClose, SYS_PREFIX "sys_close");
 }
 
+void DiscoveryBpf::attachOpenSSLProbesToLibName(const std::string& libName) {
+	skel->links.uprobeSSLReadOpenSSL = attachUprobeToLibFunc(skel->progs.uprobeSSLReadOpenSSL, libName, "SSL_read");
+	skel->links.uretprobeSSLReadOpenSSL = attachUretprobeToLibFunc(skel->progs.uretprobeSSLReadOpenSSL, libName, "SSL_read");
+	skel->links.uprobeSSLReadExOpenSSL = attachUprobeToLibFunc(skel->progs.uprobeSSLReadExOpenSSL, libName, "SSL_read_ex");
+	skel->links.uretprobeSSLReadExOpenSSL = attachUretprobeToLibFunc(skel->progs.uretprobeSSLReadExOpenSSL, libName, "SSL_read_ex");
+	skel->links.uprobeSSLPendingOpenSSL = attachUprobeToLibFunc(skel->progs.uprobeSSLPendingOpenSSL, libName, "SSL_pending");
+	skel->links.uretprobeSSLPendingOpenSSL = attachUretprobeToLibFunc(skel->progs.uretprobeSSLPendingOpenSSL, libName, "SSL_pending");
+}
+
+void DiscoveryBpf::attachOpenSSLProbes() {
+	auto libNames = {"libssl.so", "libssl3.so", "libssl.so.3", "libssl1.so", "libssl.so.1"};
+	for (const auto& libName : libNames) {
+		try {
+			attachOpenSSLProbesToLibName(libName);
+			return; // success
+		} catch (const std::runtime_error& e) {
```

Comment:
throw - catch should be used for error handling. Here it's not error path, but standard execution. Warning is redundant